### PR TITLE
[Refactoring] Replace `ChipTool` with `ChipServer` and `MatterYAMLRunner`

### PR DESCRIPTION
**Pull request by ccruzagralopes**
Next step on the refactoring task:
- https://github.com/project-chip/certification-tool/issues/95

## What changed

- Created the `ChipServer` class to handle operations related to the `chip-tool` and `chip-app1` binaries.
- Created the `MatterYAMLRunner ` class to handle operations related to running a YAML test case and commissioning.

## Testing

I ran one of each test case type and they were all ok.

Screenshot 2024-01-15 at 19 13 34
Screenshot 2024-01-15 at 19 13 42
Screenshot 2024-01-15 at 19 13 48

## Warning

There are changes in the python_tests submodule. So, before this PR can be merged, the following one must be merged and then the submodule SHA must be updated:
- https://github.com/project-chip/matter-test-scripts/pull/104

### CODE REVIEW

File: test_collections/sdk_tests/support/chip/chip_server.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
+#
+# Copyright (c) 2023 Project CHIP Authors
```
Comment (rquidute) [inline test_collections/sdk_tests/support/chip/chip_server.py:2]:
```suggestion
# Copyright (c) 2024 Project CHIP Authors
```

**Review by rquidute — APPROVED** [review]

**Review by antonio-amjr — APPROVED** [review]

File: test_collections/sdk_tests/support/tests/yaml_tests/test_matter_yaml_runner.py [hunk 2]
```diff
@@ -0,0 +1,460 @@
+#
+# Copyright (c) 2023 Project CHIP Authors
```
Comment (ccruzagralopes) [inline test_collections/sdk_tests/support/tests/yaml_tests/test_matter_yaml_runner.py:2]:
```suggestion
# Copyright (c) 2024 Project CHIP Authors
```

**Review by fabiowmm — COMMENTED** [review]
Just a little comment.

File: test_collections/sdk_tests/support/yaml_tests/models/chip_suite.py [hunk 3]
```diff
@@ -56,23 +58,23 @@ async def setup(self) -> None:
...
+            self.runner.set_pics(pics=self.pics)
         else:
-            # Disable sending "-PICS" option when running chip server
-            self.chip_tool.reset_pics_state()
+            # Disable sending "-PICS" option when running test
+            self.runner.reset_pics_state()
 
         self.__dut_commissioned_successfully = False
-        if self.test_type == ChipTestType.CHIP_TOOL:
+        if self.test_type == ChipServerType.CHIP_TOOL:
```
[the remark this answers is not in the document]
Comment (ccruzagralopes) [reply test_collections/sdk_tests/support/yaml_tests/models/chip_suite.py]: I agree. I've updated the variable to `server_type`

**Review by fabiowmm — APPROVED** [review]

**Pull request merged**